kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/InMemoryCommitCoordinator.scala

```diff
@@ -0,0 +1,366 @@
+    val commitResponse = commitImpl(
+      logStore,
+      hadoopConf,
+      logPath,
+      coordinatedCommitsTableConf,
+      commitVersion,
+      fileStatus,
+      commitTimestamp)
+
+    val mcToFsConversion = isCoordinatedCommitsToFSConversion(commitVersion, updatedActions)
+    // Backfill if needed
+    if (batchSize <= 1) {
+      // Always backfill when batch size is configured as 1
+      backfill(logStore, hadoopConf, logPath, commitVersion, fileStatus)
+      val targetFile = CoordinatedCommitsDefaultUtils.getHadoopDeltaFile(logPath, commitVersion)
+      val targetFileStatus = fs.getFileStatus(targetFile)
+      val newCommit = commitResponse.getCommit.withFileStatus(targetFileStatus)
+      return new CommitResponse(newCommit)
+    }
+    else if (commitVersion % batchSize == 0 || mcToFsConversion) {
+      logger.info(
+        "Making sure commits are backfilled till {} version for table {}",
+        commitVersion,
+        tablePath)
+      backfillToVersion(
+        logStore,
+        hadoopConf,
+        logPath,
+        coordinatedCommitsTableConf,
+        commitVersion,
+        null)
+    }
+    logger.info("Commit {} done successfully on table {}", commitVersion, tablePath)
+    commitResponse
+  }
+
+  override def getCommits(
+    logPath: Path,
+    coordinatedCommitsTableConf: util.Map[String, String],
+    startVersion: lang.Long,
+    endVersion: lang.Long): GetCommitsResponse = withReadLock[GetCommitsResponse](logPath) {
+    val tableData = perTableMap.get(logPath.toString)
+    val startVersionOpt = Optional.ofNullable(startVersion)
+    val endVersionOpt = Optional.ofNullable(endVersion)
+    val effectiveStartVersion = startVersionOpt.orElse(0L)
+    // Calculate the end version for the range, or use the last key if endVersion is not
+    // provided
+    val effectiveEndVersion = endVersionOpt.orElseGet(
+      () => if (tableData.commitsMap.isEmpty) effectiveStartVersion
+    else tableData.commitsMap.lastKey)
+    val commitsInRange = tableData.commitsMap.subMap(effectiveStartVersion, effectiveEndVersion + 1)
+    new GetCommitsResponse(
+      new util.ArrayList[Commit](commitsInRange.values), tableData.lastRatifiedCommitVersion)
+  }
+
+  override def backfillToVersion(
+    logStore: LogStore,
+    hadoopConf: Configuration,
+    logPath: Path,
+    coordinatedCommitsTableConf: util.Map[String, String],
+    version: Long,
+    lastKnownBackfilledVersion: lang.Long): Unit = {
+    // Confirm the last backfilled version by checking the backfilled delta file's existence.
+    var validLastKnownBackfilledVersion = lastKnownBackfilledVersion
+    if (lastKnownBackfilledVersion != null) try {
+      val fs = logPath.getFileSystem(hadoopConf)
+      if (!fs.exists(CoordinatedCommitsDefaultUtils.getHadoopDeltaFile(logPath, version))) {
+        validLastKnownBackfilledVersion = null
+      }
+    } catch {
+      case e: IOException =>
```

Comment:
Since the top-level interface already declares that this method can throw IOException, you don't need to catch it here.